src/liftdrag_plugin/advanced_liftdrag_plugin.cpp

```diff
@@ -0,0 +1,641 @@
+      lift_force_pub_ = node_handle_->Advertise<physics_msgs::msgs::Force>("~/" + lift_force_topic);
+      gzdbg << "Publishing to ~/" << lift_force_topic << std::endl;
+  }
+
+  if (_sdf->HasElement("windSubTopic")){
+    this->wind_sub_topic_ = _sdf->Get<std::string>("windSubTopic");
+    wind_sub_ = node_handle_->Subscribe("~/" + wind_sub_topic_, &AdvancedLiftDragPlugin::WindVelocityCallback, this);
+  }
+
+}
+
+/////////////////////////////////////////////////
+void AdvancedLiftDragPlugin::OnUpdate()
+{
+  GZ_ASSERT(this->link, "Link was NULL");
+  // get linear velocity at ref_pt in inertial frame
+#if GAZEBO_MAJOR_VERSION >= 9
+  ignition::math::Vector3d vel = this->link->WorldLinearVel(this->ref_pt) - wind_vel_;
+  const common::Time current_time = this->world->SimTime();
+#else
+  ignition::math::Vector3d vel = ignitionFromGazeboMath(this->link->GetWorldLinearVel(this->ref_pt)) - wind_vel_;
+  const common::Time current_time = this->world->GetSimTime();
+#endif
+  const double dt = (current_time - this->last_pub_time).Double();
+
+  if (vel.Length() <= 0.01)
+    return;
+
+  // pose of body
+#if GAZEBO_MAJOR_VERSION >= 9
+  ignition::math::Pose3d pose = this->link->WorldPose();
+#else
+  ignition::math::Pose3d pose = ignitionFromGazeboMath(this->link->GetWorldPose());
+#endif
+
+  //Define body frame: X forward, Z downward, Y out the right wing
+  ignition::math::Vector3d bodyX = pose.Rot().RotateVector(this->forward);
+  ignition::math::Vector3d bodyZ = -1*(pose.Rot().RotateVector(this->upward));
+  ignition::math::Vector3d bodyY =  bodyZ.Cross(bodyX);
+
+  // Get the in-plane velocity (remove spanwise velocity from the velocity vector vel)
+  ignition::math::Vector3d velInLDPlane = vel - vel.Dot(bodyY)*bodyY;
+
+  // Define stability frame: X is in-plane velocity, Y is the same as body Y,
+  // and Z perpendicular to both
+  ignition::math::Vector3d stabilityX = velInLDPlane;
+  stabilityX.Normalize();
+  ignition::math::Vector3d stabilityY = bodyY;
+  ignition::math::Vector3d stabilityZ = stabilityX.Cross(stabilityY);
+
+  double span = std::sqrt(this->area*this->AR); // Wing span
+  double mac = this->area/span; //Mean aerodynamic chord
+
+  //Get non-dimensional body rates. Gazebo uses ENU, so some have to be flipped
+  ignition::math::Vector3d body_rates = this->link->RelativeAngularVel();
+  double speed = vel.Length();
+  double p = body_rates.X()*span/(2*speed); //Non-dimensionalized roll rate
+  double q = -1*body_rates.Y()*mac/(2*speed); //Non-dimensionalized pitch rate
+  double r = -1*body_rates.Z()*span/(2*speed); //Non-dimensionalized yaw rate
+
+  if (bodyX.Dot(vel) <= 0.0){
+    // Only calculate lift or drag if the wind relative velocity is in the same direction
+    return;
+  }
+
+  //Compute angle of attack, alpha, using the stability and body axes
+  double stabXHoriz = stabilityX.Dot(bodyX);
+  double stabXVert = stabilityX.Dot(bodyZ);
```

Comment:
again just recommendation - maybe putting the word "axis" there makes it a bit more clear what these variables are.

Reply:
Thanks. Hopefully these are clearer now (projections of stability axes onto body axes)